packages/framework-types/src/envelope.ts

```diff
@@ -83,12 +84,24 @@ export interface ReadModelRequestEnvelope extends Envelope {
   paginatedVersion?: boolean // Used only for retrocompatibility
 }
 
+export interface ReadModelByIdRequestEnvelope extends Envelope {
+  typeName: string
+  version: number
+  id: string
+  sequenceKey?: SequenceKey
+}
+
 export interface ReadModelRequestArgs {
   filter?: ReadModelRequestProperties
   limit?: number
   afterCursor?: unknown
 }
 
+export interface ReadModelByIdRequestArgs {
+  id: string
+  [sequenceKey: string]: string | undefined
```

Comment:
This is a [index signature](https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures), not a declaration for a single field. Not sure if this is what you meant, this means:

> ReadModelByIdRequestArgs has multiple fields that have a `string` name, and will have of value `string | undefined`

Doesn't matter too much as it works, but take into account that this would accept any kind of field, even multiple ones